packages/connect-thegraph/src/connector.ts

```diff
@@ -101,19 +104,31 @@ class ConnectorTheGraph implements IOrganizationConnector {
     organization: Organization,
     appAddress: Address
   ): Promise<Role[]> {
-    return this.#gql.performQueryWithParser<Role[]>(
-      queries.ROLE_BY_APP_ADDRESS('query'),
-      { appAddress: appAddress.toLowerCase() },
-      (result) => parseRoles(result, organization)
-    )
+    try {
+      return this.#gql.performQueryWithParser<Role[]>(
+        queries.ROLE_BY_APP_ADDRESS('query'),
+        { appAddress: appAddress.toLowerCase() },
+        async (result) => parseRoles(result, organization)
+      )
+    } catch (err) {
+      throw new ErrorUnexpectedResult(
+        'Unexpected result when fetching the roles.'
```

Comment:
I think we should pass the `err` as the `reason` option otherwise we swallow the errors from the `GraphQLWrapper` don't we?

Reply:
Yes this could definitely be an issue.  I think it could be a problem if we start using `reason` for that, as it is supposed to be a string that helps to distinguish between two errors of the same type. We might also have a chain of errors, I wonder if it would be useful to provide a way to users so they can explore it? Or if we should let this to the debugger. :thinking:

I was discussing with @sohkai about `reason`, and he was making the point that it shouldn’t be that useful, as apps probably won’t act differently based on the `reason`, but on the type only. For example, if an `ErrorConnection` fails for X or Y reason, there shouldn’t be a need for the app to do anything else than retrying to connect. Authors will also have the error message that will be specific and provide context. If making the distinction between two errors of the same type really matters, then it might be a sign that we need to extend the error (which is more or less the role `reason` is having anyway, by acting as a “light subtype”).

Now for this specific case: I think we could suggest to enable the verbosity in the message itself, so authors can see why the query failed and debug it? Something related: I would like to discuss the possibility to have different levels of verbosity rather than a global verbosity flag, having this would also help here.

What do you think?

Reply:
Update: I removed `reason`.